### ISSUE TITLE
Feature request: send message in groupchat

### ISSUE THREAD

**Issue by joshcoales** [JosXa/tgintegration#7]
* tgintegration version: 0.16.0
* Python version: 3.6.3
* Operating System: Windows

### Description

It would be very handy to be able to test that the bot responds correctly to a message in a groupchat. Currently this library only tests against the bot in privmsg

### What I Did

Tried feeding chat_id parameter to send_message_await(), but it does not like that, as it's overridden to the bot id

**Comment by JosXa** [comment]
You want to use the base client for that, which is not bound to a specific chat_id. Come talk to me at https://t.me/TgIntegration

**Comment by JosXa** [comment]
While this is technically already possible, I agree that it is a good idea to implement this directly in the library.

See, since v1.0 you can do the following:

``` python
# Controller reacts to messages from @deerspangle
controller = BotController(peer="@deerspangle", client=client)

# This internally merges the controller's `filters.user("@deerspangle")` 
# with `filters.chat("@TgIntegration")` as an AND relation
async with controller.collect(filters.chat("@TgIntegration")) as response:
    await client.send_message("@TgIntegration", "Hello there!")

assert response.messages[0].chat.username == "TgIntegration"
assert response.messages[0].user.username == "deerspangle"
```

Now, I see the need for making tests in groups and channels a high-level concept and would love some ideas on what a good user-facing API for that would be.
I'm thinking something like

```
controller = BotController(
    peer="@mybot",  # bot under test
    client=client,
    context=InteractionContext(
        username="@somechat",
        type_="group",
        title="Yo I'm just testing man",
        create_if_not_exists=True  # user clients ftw 💪 
    )
)
```

However, context is maybe too generic? And also I'm debating if it should be a dataclass or just a bunch of kwargs passed to the initializer...

**Comment by joshcoales** [comment]
I like being able to set the context with a context manager (It's in the name, right?
As it means you can test things like: Set something in private message with the bot, then test it is applied in chat. Which seems potentially very useful!

**Comment by JosXa** [comment]
Oh yes, that's an angle I haven't thought about, you're right!

**Comment by JosXa** [comment]
Only drawback would be that tests could potentially grow really wide with all those contextanagers..? Like you are always at least two levels deep

**Comment by joshcoales** [comment]
That is true, but feels like it's a rare enough case that people can cope with being tabbed in a few times at that point. Or even extract out parts of their test.
